### PR TITLE
disable region detail logs in kv-client by default

### LINKED ISSUE #10486: Region logs are too noisy in module kv-client
### Before asking a question, make sure you have

- [X] Searched existing Stack Overflow questions
- [X] Googled your question
- [X] Searched open and closed [GitHub issues](https://github.com/pingcap/tiflow/issues?q=is%3Aissue)
- [X] Read the [documentation](https://docs.pingcap.com/tidb/stable)

### What is your question?

There is a known problem that there are too many region detail logs in module kv-client. Generally they are not helpful in most cases.
So we can disable them by default. If we meet issues about resolved timestamp, then we can re-enable them with a configuration.

### PR DESCRIPTION
**Pull request by hicqu**
### What problem does this PR solve?

Issue Number: close #10486 

### What is changed and how it works?

* Disable region detail logs in kv-client by default;
* Add a new debug option to enable those logs.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by sdojjy — APPROVED** [review]

File: pkg/config/debug.go [hunk 1]
```diff
@@ -60,4 +60,6 @@ type PullerConfig struct {
 	EnableResolvedTsStuckDetection bool `toml:"enable-resolved-ts-stuck-detection" json:"enable-resolved-ts-stuck-detection"`
 	// ResolvedTsStuckInterval is the interval of checking resolved ts stuck.
 	ResolvedTsStuckInterval TomlDuration `toml:"resolved-ts-stuck-interval" json:"resolved-ts-stuck-interval"`
+	// LogRegionDetails determines whether logs Region details or not in puller and kv-client.
```
Comment (CharlesCheung96) [inline pkg/config/debug.go:63]: Perhaps it would be more appropriate to use changefeed level parameters and support online updates for LogRegionDetails?

Comment (hicqu) [reply pkg/config/debug.go:63]: Just a server level config is fine enough.

Comment (asddongmen) [reply pkg/config/debug.go:63]: @hicqu @CharlesCheung96 TiCDC has already supported online log level changing. So we can change log level to debug level to print these logs when needed.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**